build/lin.sh

```diff
@@ -155,12 +175,15 @@ cd ${DEPS}/zlib
 ./configure --prefix=${TARGET} ${LINUX:+--uname=linux} ${DARWIN:+--uname=darwin} --static
 make install
 
+# Do not build ffi on darwin-arm64v8 so that meson will build it itself properly
+if [ $DARWIN_ARM != true ]; then
```

Comment:
This appears to make meson fall back to a "wrapped" https://gitlab.freedesktop.org/gstreamer/meson-ports/libffi fork, which is not quite the same as libffi v3.3 and also prevents control over the `--disable-*` build flags we're specifying. Is this condition still required?

Reply:
> also prevents control over the --disable-* build flags 

Oh, I totally forgot about those "--disable". That's a problem.

Making meson fall back on a wrapped libffi fork is by design, as the mesonified libffi happens to build properly while trunk libffi does not.
Unfortunately, libffi didn't tag a release in a long while, so the condition is still required.

That said, now that I know you're interested in the PR, I'll look into solutions! I'm pretty sure the homebrew/macports people have working buildscripts.